strr-host-pm-web/app/pages/dashboard-new/index.vue

```diff
@@ -0,0 +1,328 @@
+<script setup lang="ts">
+const { t } = useI18n()
+const localePath = useLocalePath()
+const accountStore = useConnectAccountStore()
+const strrModal = useStrrModals()
+const { deleteApplication, getAccountRegistrations, getAccountApplications } = useStrrApi()
+const { isDashboardTableSortingEnabled, isNewDashboardEnabled } = useHostFeatureFlags()
+
+// Redirect to old dashboard if feature flag is not enabled
+if (!isNewDashboardEnabled.value) {
+  await navigateTo(localePath('/dashboard'))
+}
+
+// Applications list setup
+const applicationsLimit = ref(6)
+const applicationsPage = ref(1)
+
+// Define allowed statuses for applications in progress
+const allowedStatuses = [
+  ApplicationStatus.DRAFT,
+  ApplicationStatus.PAYMENT_DUE,
+  ApplicationStatus.PAID,
+  ApplicationStatus.ADDITIONAL_INFO_REQUESTED,
+  ApplicationStatus.FULL_REVIEW,
+  ApplicationStatus.DECLINED,
+  ApplicationStatus.PROVISIONALLY_DECLINED,
+  ApplicationStatus.PROVISIONAL,
+  ApplicationStatus.NOC_PENDING,
+  ApplicationStatus.NOC_EXPIRED,
+  ApplicationStatus.PROVISIONAL_REVIEW_NOC_PENDING,
+  ApplicationStatus.PROVISIONAL_REVIEW_NOC_EXPIRED
+]
+
+// Registrations list setup
+const registrationsLimit = ref(6)
+const registrationsPage = ref(1)
+
+useHead({
+  title: t('page.dashboardList.title')
+})
+
+definePageMeta({
+  middleware: ['auth', 'check-tos', 'require-account']
+})
+
+setBreadcrumbs([
+  {
+    label: t('label.bcregDash'),
+    to: useRuntimeConfig().public.registryHomeURL + 'dashboard',
+    appendAccountId: true,
+    external: true
+  },
+  { label: t('page.dashboardList.h1') }
+])
+
+// Helper to create sortable column
+const createColumn = (key: string, label: string, sortable = true) => ({
+  key,
+  label,
+  ...(sortable && { sortable: isDashboardTableSortingEnabled.value })
+})
+
+const commonColumns = [
+  createColumn('number', t('label.number')),
+  createColumn('status', t('label.status')),
+  createColumn('address', t('label.address')),
+  createColumn('localGovernment', t('label.localGovernment'))
+]
+
+const applicationsColumns = [
+  ...commonColumns,
+  createColumn('dateSubmitted', t('label.dateSubmitted')),
+  createColumn('actions', t('label.actions'), false)
+]
+
+const registrationsColumns = [
+  ...commonColumns,
+  createColumn('expiryDate', t('label.expirationDate')),
+  createColumn('actions', t('label.actions'), false)
+]
+
+// Fetch Applications
+const { data: applicationsResp, status: applicationsStatus, refresh: refreshApplications } = await useAsyncData(
```

Comment:
I would put fetching applications and registrations into it's own components, so they can be loaded on its own without blocking the page.

Reply:
Done